From 31094eefe9bfc69f40e814a367675d212496c989 Mon Sep 17 00:00:00 2001
From: Niall Pemberton <niallp@apache.org>
Date: Sun, 31 Jan 2010 17:24:49 +0000
Subject: [PATCH] Port LANG-467 to 2.x branch - Rolling back r611543 for
 LANG-393, and removing the special handling of BigDecimal to use compareTo
 instead of equals because it creates an inequality with HashCodeBuilder

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/lang/branches/LANG_2_X@905078 13f79535-47bb-0310-9956-ffa450edef68
---
 .../org/apache/commons/lang/builder/EqualsBuilder.java    | 8 ++------
 .../apache/commons/lang/builder/EqualsBuilderTest.java    | 8 --------
 2 files changed, 2 insertions(+), 14 deletions(-)

diff --git a/src/main/java/org/apache/commons/lang/builder/EqualsBuilder.java b/src/main/java/org/apache/commons/lang/builder/EqualsBuilder.java
index 50fe3121e86..100180640bc 100644
--- a/src/main/java/org/apache/commons/lang/builder/EqualsBuilder.java
+++ b/src/main/java/org/apache/commons/lang/builder/EqualsBuilder.java
@@ -375,12 +375,8 @@ public EqualsBuilder append(Object lhs, Object rhs) {
         }
         Class lhsClass = lhs.getClass();
         if (!lhsClass.isArray()) {
-            if (lhs instanceof java.math.BigDecimal && rhs instanceof java.math.BigDecimal) {
-                isEquals = (((java.math.BigDecimal)lhs).compareTo((java.math.BigDecimal)rhs) == 0);
-            } else {
-                // The simple case, not an array, just test the element
-                isEquals = lhs.equals(rhs);
-            }
+            // The simple case, not an array, just test the element
+            isEquals = lhs.equals(rhs);
         } else if (lhs.getClass() != rhs.getClass()) {
             // Here when we compare different dimensions, for example: a boolean[][] to a boolean[] 
             this.setEquals(false);
diff --git a/src/test/java/org/apache/commons/lang/builder/EqualsBuilderTest.java b/src/test/java/org/apache/commons/lang/builder/EqualsBuilderTest.java
index 710f0bdad42..e5fc9042c8d 100644
--- a/src/test/java/org/apache/commons/lang/builder/EqualsBuilderTest.java
+++ b/src/test/java/org/apache/commons/lang/builder/EqualsBuilderTest.java
@@ -377,14 +377,6 @@ public void testFloat() {
         assertTrue(new EqualsBuilder().append(Float.POSITIVE_INFINITY, Float.POSITIVE_INFINITY).isEquals());
     }
 
-    // https://issues.apache.org/jira/browse/LANG-393
-    public void testBigDecimal() {
-        BigDecimal o1 = new BigDecimal("2.0");
-        BigDecimal o2 = new BigDecimal("2.00");
-        assertTrue(new EqualsBuilder().append(o1, o1).isEquals());
-        assertTrue(new EqualsBuilder().append(o1, o2).isEquals());
-    }
-
     public void testAccessors() {
         EqualsBuilder equalsBuilder = new EqualsBuilder();
         assertTrue(equalsBuilder.isEquals());
